Completed Timer

VictorYao17 opened the pull request:
All classes related to timer are working and completed with JavaDocs

madelahn reviewed (changes requested):
Your implementations seem great, Victor! Make sure to work on the implementations of scaling gold with the timer so that the user can get more money to buy items.
Also, when you get around to it, be sure to create all your tests, and delete anything that isn't being used - rather than commenting it out.

AVLTG reviewed (approved):
These are really good changes! I have to say, however, looking through your code, I realize how as a team we all collectively made a silly mistake. Whenever you have delete item delete an item from your inventory, the deleted item will show up again within the delete item menu after you leave it an re-enter it again, because the file is not being written correctly. public void save() in PLayerInventoryFile.java is probably not saving the file correctly with what you want it to, so make sure you check that out and try to fix it, as this probably creates issues for other features other than deleteitem too like for the shopinventory or equipping in playerinventory. Other than that looks like everything with java swing works though!!! Good job!!

AVLTG reviewed (changes requested):
Accidentally approved merging when I meant to request changes

VictorYao17 commented:
I have added and completed tests for each testable class. The tests for ActionListeners for the buttons on the UI are passing in IntelliJ but failing on GitHub and I am not sure how to fix it. I have fixed all of the dependency violations and resolved all warnings as well. Everything should be good to be merged into the main branch.

jfongli reviewed (approved):
I think this looks great, thanks for your hard work! 
About the GitHub tests- I think they're to be expected in this situation (at least that's what I understood from an answer on @\632 on piazza).

jahink17 reviewed (approved):
Looks good!

jahink17 reviewed (approved):
👍

jfongli reviewed (approved):
Same as my last comment haha. Thanks again!

m1k2l3 reviewed (approved):
Great work Victor!
